src/components/Firestore/Requests/RequestDetails/InspectionSection/InspectionBlock.tsx

```diff
@@ -0,0 +1,84 @@
+/**
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import './InspectionBlock.scss';
+
+import { Icon } from '@rmwc/icon';
+import { ThemeProvider } from '@rmwc/theme';
+import classnames from 'classnames';
+import React, { useState } from 'react';
+
+import { darkestGray } from '../../../../../colors';
+
+const INSPECT_BLOCK_CLASS = 'Firestore-Request-Details-Inspection-Block';
+const MAIN_INSPECT_BLOCK_CLASS = INSPECT_BLOCK_CLASS + '--Main';
+
+interface Props {
+  label: string;
+  value?: string;
+  isMainBlock?: boolean;
+}
+
+export const InspectionBlock: React.FC<Props> = ({
+  label,
+  value,
+  isMainBlock,
+  children,
+}) => {
+  // Main inspection-blocks start expanded by default
+  const [isExpanded, setIsExpanded] = useState<Boolean>(!!isMainBlock);
+
+  // NOTE: the content of the InspectionBlock could be either a string
+  // (passed through the value property) or HTML children elements.
+  const renderContent = () => {
+    if (children) {
+      return children;
+    }
+    // TODO: improve how the value property is rendered: differ by value types
+    //       (string, number, boolean, object and maybe array too)
+    return (
+      <div className={`${INSPECT_BLOCK_CLASS}-Value`} title={value}>
+        {value}
+      </div>
+    );
+  };
+
+  return (
+    <ThemeProvider
+      options={{
+        inspectionMainBlockBackground: darkestGray,
```

Comment:
Don't extend the `--mdc-theme-` namespace to avoid confusion. Is there a color in material theme that is close enough to your desired gray? If not, just use a color value in CSS and avoid `ThemeProvider` (since this component won't respond to theming at all).

Reply:
There is no existing color that is close enough to the desired gray, I actually came up with a solution using `ThemeProvider`, since the inspection-block can either be a main-block (which has a darker grey as background) or a regular-block (which uses the existing `grey100` as background).

Let me know what you think.

![image](https://user-images.githubusercontent.com/34299428/105773587-ba5eb980-5f29-11eb-8b62-1eb787f629f9.png)
![image](https://user-images.githubusercontent.com/34299428/105773618-c9456c00-5f29-11eb-8c3f-44317780b2f5.png)

Reply:
Sure, let's go with your latest proposed approach.